saleor/graphql/webhook/subscription_query.py

```diff
@@ -48,22 +48,44 @@ def validate_query(self) -> list[Union[GraphQLSyntaxError, ValidationError]]:
             self.error_code = WebhookErrorCode.GRAPHQL_ERROR.value
             return errors
 
+        subscription = self._get_subscription(self.ast)
+        if not subscription:
+            self.error_code = WebhookErrorCode.MISSING_SUBSCRIPTION.value
+            return [
+                ValidationError(
+                    message="Subscription operation can't be found.",
+                    code=WebhookErrorCode.MISSING_SUBSCRIPTION.value,
+                )
+            ]
+
+        field_names = [
+            selection.name.value for selection in subscription.selection_set.selections
+        ]
+        if len(field_names) > 1 and set(field_names) != {"event"}:
+            self.error_code = WebhookErrorCode.INVALID.value
+            return [
+                ValidationError(
+                    message="Subscription must select only one top field.",
+                    code=WebhookErrorCode.INVALID.value,
+                )
+            ]
+
         try:
-            self.events = self.get_events_from_subscription()
+            self.events = self.get_events_from_subscription(subscription)
         except ValidationError as err:
             self.error_code = err.code
             return [err]
 
         self.is_valid = True
         return []
 
-    def get_events_from_subscription(self) -> list[str]:
-        subscription = self._get_subscription(self.ast)
-        if not subscription:
-            raise ValidationError(
-                message="Subscription operation can't be found.",
-                code=WebhookErrorCode.MISSING_SUBSCRIPTION.value,
-            )
+    def get_events_from_subscription(self, subscription) -> list[str]:
+        subscription_name = subscription.selection_set.selections[0].name.value
```

Comment:
Are you sure that you will always have something in `subscription.selection_set.selections`?

Reply:
I tested this, and I wasnt able to get something different. Additionally we already fetch selections in this way, since beginning of 2023
![image](https://github.com/user-attachments/assets/5ca411ad-a330-45d9-a650-c9696f299a51)